src/d/actor/d_a_obj_eff.cpp

```diff
@@ -2,261 +2,479 @@
 // Generated by dtk
 // Translation Unit: d_a_obj_eff.cpp
 //
-
 #include "d/actor/d_a_obj_eff.h"
 #include "d/d_procname.h"
 #include "d/d_priority.h"
+#include "d/d_com_inf_game.h"
+
+#include "weak_bss_936_to_1036.h" // IWYU pragma: keep
+#include "weak_data_1811.h" // IWYU pragma: keep
+#include "weak_data_2100_2080.h"
+
+const GXColor daObjEff::BarrelSmokeCB::original_color = {0xAF, 0x96, 0x64, 0xFF};
+
 
 /* 000000EC-00000140       .text __ct__Q28daObjEff13BarrelSmokeCBFv */
-daObjEff::BarrelSmokeCB::BarrelSmokeCB() {
-    /* Nonmatching */
+daObjEff::BarrelSmokeCB::BarrelSmokeCB() : dPa_smokeEcallBack(original_color, NULL, 0) {
+    field_0x20 = 0x3c;
 }
 
 /* 00000140-000001CC       .text execute__Q28daObjEff13BarrelSmokeCBFP14JPABaseEmitter */
-void daObjEff::BarrelSmokeCB::execute(JPABaseEmitter*) {
-    /* Nonmatching */
+void daObjEff::BarrelSmokeCB::execute(JPABaseEmitter *i_emitter) {
+    field_0x20 -= 1;
+    if (field_0x20 <= 0) {
+        end();
+    } else if (field_0x20 < 0x32) {
+        i_emitter->mGlobalPrmColor.a = field_0x20 * 3.6f;
```

Comment:
Debug map says this inline is used
```suggestion
        i_emitter->setGlobalAlpha(field_0x20 * 3.6f);
```